lightning/src/ln/channel.rs

```diff
@@ -1821,7 +1821,33 @@ where
 				res
 			},
 			ChannelPhase::Funded(mut funded_channel) => {
-				let res = funded_channel.commitment_signed(msg, logger).map(|monitor_update_opt| (None, monitor_update_opt));
+				#[cfg(splicing)]
+				let has_negotiated_pending_splice = funded_channel.pending_splice.as_ref()
+					.map(|pending_splice| pending_splice.funding.is_some())
+					.unwrap_or(false);
+				#[cfg(splicing)]
+				let session_received_commitment_signed = funded_channel
+					.interactive_tx_signing_session
+					.as_ref()
+					.map(|session| session.has_received_commitment_signed());
+				#[cfg(splicing)]
+				let res = if has_negotiated_pending_splice
+					// Not having a signing session implies they've already sent `splice_locked`,
+					// which must always come after the initial commitment signed is sent.
+					&& !session_received_commitment_signed.unwrap_or(true)
```

Comment:
nit: `.unwrap_or(true)` can be called when initializing `session_received_commitment_signed` like `has_negotiated_pending_splice`.